app/frontend/components/book_selection_menu.js

```diff
@@ -123,33 +123,93 @@ class BookSelectionMenu {
     }
   }
 
-  async selectBibleBook(book_code, book_title, reference_book_title) {
-    var currentBibleTranslationId = app_controller.tab_controller.getTab().getBibleTranslationId();
-    if (currentBibleTranslationId == null || currentBibleTranslationId == undefined) {
+  async selectBibleBook(bookCode, bookTitle, referenceBookTitle) {
+    this.currentBibleTranslationId = app_controller.tab_controller.getTab().getBibleTranslationId();
+    if (this.currentBibleTranslationId == null || this.currentBibleTranslationId == undefined) {
       return;
     }
 
     Sentry.addBreadcrumb({category: "app",
-                          message: `Selected book ${book_code} using translation ${currentBibleTranslationId}`,
+                          message: `Selected book ${bookCode} using translation ${this.currentBibleTranslationId}`,
                           level: Sentry.Severity.Info});
     
-    var books = await ipcNsi.getBookList(currentBibleTranslationId);
-    if (!books.includes(book_code)) {
+    var books = await ipcNsi.getBookList(this.currentBibleTranslationId);
+    if (!books.includes(bookCode)) {
       return;
     }
 
+    const selectChapterBeforeLoading = app_controller.optionsMenu._selectChapterBeforeLoadingOption;
+    const bookChapterCount = await ipcNsi.getBookChapterCount(this.currentBibleTranslationId, bookCode);
+
+    if (selectChapterBeforeLoading.isChecked && bookChapterCount > 1) {
+      //console.log(`Showing chapter list for ${bookTitle} ` +
+      //            `since its chapter count (${bookChapterCount}) is above the limit for instant loading!`);
+      
+      var menuBookList = document.getElementById('book-selection-menu-book-list');
```

Comment:
maybe use let/const to limit scope of this variable to the current if-block (if this is the case)